sunspot/lib/sunspot/field.rb

```diff
@@ -235,6 +231,11 @@ def eql?(field)
     end
 
     alias_method :==, :eql?
+
+    def target
+      @target = Util.full_const_get(@target) if @target.is_a?(String)
```

Comment:
Is this going to keep loading and assigning at each call? Like should this be `@target ||= ...` instead?

Reply:
Only for the first time. The `@target` will then change from `"Public::Profile"` to `Public::Profile`

Reply:
Got it, makes sense.